src/App.tsx

```diff
@@ -1,24 +1,65 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
+import { Clock } from './components/Clock/Clock';
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean,
+  clockName: string | number,
 };
 
+class App extends Component<{}, State> {
+  state = {
+    isClockVisible: true,
+    clockName: 'Magic clock',
+  };
+
+  componentDidUpdate(_prevState: {}, prevProps: State) {
+    // eslint-disable-next-line
+    console.log(`The Clock was renamed from ${prevProps.clockName} to ${this.state.clockName}`);
+  }
```

Comment:
Move that logic to a Clock component, to make it work properly, cause now I get this message even when I press show/hide button